src/client/common/experiments.ts

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { inject, injectable, named } from 'inversify';
+import { Uri } from 'vscode';
+import { IHttpClient } from '../activation/types';
+import { sendTelemetryEvent } from '../telemetry';
+import { EventName } from '../telemetry/constants';
+import { IApplicationEnvironment, IWorkspaceService } from './application/types';
+import { STANDARD_OUTPUT_CHANNEL } from './constants';
+import { traceDecorators, traceError } from './logger';
+import { ICryptoUtils, IExperimentsManager, IOutputChannel, IPersistentState, IPersistentStateFactory, Resource } from './types';
+import { Experiments } from './utils/localize';
+
+const EXPIRY_DURATION_MS = 30 * 60 * 1000;
+const experimentStorageKey = 'EXPERIMENT_STORAGE_KEY';
+const configUri = 'https://raw.githubusercontent.com/karrtikr/check/master/environments.json';
+
+type AB_Experiments = { name: string; salt: string; min: number; max: number }[];
```

Comment:
Please remove the `_`, TypeScript doesn't use `_` in type names.
```suggestion
type ABExperiments = { name: string; salt: string; min: number; max: number }[];
```